### PR TITLE
Fzf: init module

### PR DESCRIPTION
**Pull request by laalsaas**
###### Description of changes
Add fzf module to automatically source completion and keybindings in global {bash,zsh}rc. No fish, since this is already done by the package. cleanup the package a bit.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [xJ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/programs/fzf.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      keybindings = mkOption {
+        type = types.bool;
+        description = lib.mdDoc "Whether to set up fzf keybindings";
+        default = false;
+        example = true;
+      };
+    };
+  };
+  config = {
+    environment.systemPackages = if cfg.keybindings || cfg.fuzzyCompletion then [ pkgs.fzf ] else [];
```
Comment (SuperSandro2000) [inline nixos/modules/programs/fzf.nix:23]:
```suggestion
    environment.systemPackages = lib.optional (cfg.keybindings || cfg.fuzzyCompletion) pkgs.fzf;
```

File: pkgs/tools/misc/fzf/default.nix [hunk 2]
```diff
@@ -38,22 +45,19 @@ buildGoModule rec {
...
+
+    installManPage man/man1/fzf.1 man/man1/fzf-tmux.1
 
-    mkdir -p $man/share/man
-    cp -r man/man1 $man/share/man
+    install -D plugin/* -t $out/share/vim-plugins/${pname}/plugin
 
-    mkdir -p $out/share/vim-plugins/${pname}
-    cp -r plugin $out/share/vim-plugins/${pname}
+    #Install shell integrations
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/fzf/default.nix:55]:
```suggestion
    # Install shell integrations
```

File: pkgs/tools/misc/fzf/default.nix [hunk 3]
```diff
@@ -38,22 +45,19 @@ buildGoModule rec {
...
+    install -D shell/* -t $out/share/fzf/
+    install -D shell/key-bindings.fish $out/share/fish/vendor_functions.d/fzf_key_bindings.fish
+    mkdir -p $out/share/fish/vendor_conf.d
+    echo fzf_key_bindings > $out/share/fish/vendor_conf.d/load-fzf-key-bindings.fish
 
-    cp -R shell $out/share/fzf
     cat <<SCRIPT > $out/bin/fzf-share
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/fzf/default.nix:61]: We should add someting that this is deprecated.

**Comment by laalsaas** [comment]
> We should add someting that this is deprecated.

I think this is for a seperate PR, deprecating the package-share workaround
for all packages using it. (currently: zsh-autoenv, fzf, blesh and skim)

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**